src/main/java/cn/nukkit/Server.java

```diff
@@ -923,7 +923,8 @@ public void sendFullPlayerListData(Player player) {
     }
 
     public void sendRecipeList(Player player) {
-        player.dataPacket(CraftingManager.packet);
+        player.dataPacket(player.getProtocol().equals(PlayerProtocol.PLAYER_PROTOCOL_113) ?
+                CraftingManager.packet113 : CraftingManager.packet);
```

Comment:
wtf proč používáš equals když protokol je číslo? 😆

Reply:
Já to mám jako enum

Reply:
no však :D odkdy se enum porovnává pomocí equals

Reply:
Enums are final Objects so you can compare them with == instead of running equals().

Reply:
Lel :smile: I’ve been using them for 2 years and I still didn’t know that